sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/GenericResourcesImpl.java

```diff
@@ -88,16 +88,76 @@ public boolean checkExistence(String resourceGroupName, String resourceProviderN
 
     @Override
     public boolean checkExistenceById(String id) {
-        String apiVersion = getApiVersionFromId(id).block();
+        String apiVersion = getApiVersionFromIdAsync(id).block();
+        return this.checkExistenceById(id, apiVersion);
```

Comment:
Due to the `block()`, do we need to provide `checkExistenceByIdAsync` and `BeginDeleteByIdAsync`?

Reply:
`beginDeleteByIdAsync` will not be provided as this is for start of sync polling. No async until required.

`checkExistenceByIdAsync` is not there because previously the async version is not there. Any strong opinion to add?

Reply:
Just think that if no async method, it cannot be used inside a flatmap for customer. Maybe they all could wait until required.